### PR TITLE
Fix: All decks show the same number of total cards and total new cards

### LINKED ISSUE #14608: [BUG]: All decks show the same number of total cards and total new cards
### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

- Have multiple decks with different number of cards in each
- Click any deck to open the following page:

### Expected behaviour

The page for each deck shows the number of cards and new cards in that particular deck.

### Actual behaviour

All decks show the same number of cards and new cards, possibly the number of such cards in the whole collection.

### Debug info

```text
AnkiDroid Version = 2.17alpha2 (19ad510c0a44c4fa140394363a0df3bd8ab177d8)

Android Version = 13

ProductFlavor = full

Webview User Agent = Mozilla/5.0 (Linux; Android 13; moto g62 5G Build/T1SSIS33.1-75-7-1; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/118.0.0.0 Mobile Safari/537.36

ACRA UUID = d808c2e4-92ce-40db-9491-1fdcb75b1f50

Crash Reports Enabled = true

DatabaseV2 Enabled = true
```

### (Optional) Anything else you want to share?

_No response_

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
* #14608

* Perform refactorings as I see them to move towards more `suspend` functionality

## Fixes
* Fixes #14608

## Approach
* extract `queryCompletedDeckCustomStudyAction`
* use `launchCatchingTask` to execute the functionality
* move `getColUnsafe.decks.select(did)` into a `withCol`
* move `getColUnsafe.config.get("schedVer")` into `withCol`
* replace usages of `getColUnsafe.deck.selected()` with `did`
* Further reactorings for readability:
  * extract snackbar methods to inner functions
  * remove duplicated `updateUi`  method

## How Has This Been Tested?
S21, Android 13 & ran unit tests

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by mikehardy — APPROVED** [review]
Nifty - worth noting that I just checked crowdin activity stream and this is good to merge with a strings sync after, strings are clean + good to go right now, just some Ukrainian translations that appear legit

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 1]
```diff
@@ -1689,88 +1693,98 @@ open class DeckPicker :
...
+        }
 
-            if (fragmented) {
-                openStudyOptions(false)
-            } else {
-                updateDeckList()
+        if (mustUpdateScheduler) {
+            promptUserToUpdateScheduler()
+            return
+        }
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1738]: Here I think that it is easier to read if the `config` and `deck` calls are separated. As I read, it should be a guard clause like: `if the scheduler is v1 → Prompt to update and return early (i.e. don't do anything else)`

```suggestion
        if (withCol { ((config.get("schedVer") ?: 1L) == 1L) }) {
            promptUserToUpdateScheduler()
            return
        }
        
        withCol { decks.select(did) }
        // Also forget the last deck used by the Browser
        CardBrowser.clearLastDeckId()
        mFocusedDeck = did

        val deck = mDeckListAdapter.getNodeByDid(did)
        if (deck.hasCardsReadyToStudy()) {
            openReviewerOrStudyOptions(selectionType)
            return
        }
```

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 2]
```diff
@@ -2414,6 +2428,51 @@ open class DeckPicker :
...
+
+        if (decks.isDyn(did)) {
+            return@withCol CompletedDeckStatus.DYNAMIC_DECK_NO_LIMITS_REACHED
+        }
+
+        if (mDeckListAdapter.getNodeByDid(did).children.isEmpty() && isEmptyDeck(did)) {
+            return@withCol CompletedDeckStatus.EMPTY_REGULAR_DECK
+        }
+
+        return@withCol CompletedDeckStatus.REGULAR_DECK_NO_MORE_CARDS_TODAY
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:2438]:
```suggestion
        when {
            sched.hasCardsTodayAfterStudyAheadLimit() -> CompletedDeckStatus.LEARN_AHEAD_LIMIT_REACHED
            sched.newDue() || sched.revDue() -> CompletedDeckStatus.LEARN_AHEAD_LIMIT_REACHED
            decks.isDyn(did) -> CompletedDeckStatus.DYNAMIC_DECK_NO_LIMITS_REACHED
            mDeckListAdapter.getNodeByDid(did).children.isEmpty() && isEmptyDeck(did) -> CompletedDeckStatus.EMPTY_REGULAR_DECK
            else -> CompletedDeckStatus.REGULAR_DECK_NO_MORE_CARDS_TODAY
        }
```

**Review by BrayanDSO — APPROVED** [review]
Change itself LGTM. Left some refactor nits

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1738]: I was a little frustrated with the additional latency on this call compared to when we were doing it in memory. 

This is what I had originally - more than happy to move back to it

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 2 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:2438]: MUCH nicer, thanks for taking the time

**Comment by david-allison** [comment]
Changes resolved, needs a final once-over to ensure they were resolved satisfactorily

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 1 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt]: I get the feeling. I always loved micro-optimizations. Read dozens of times someone in StackOverflow telling that premature optimizations are the root of all evil.

Probably in the future we will be able to improve the background work perfomance when the repo's use of coroutines is more mature.

**Pull request merged**